app/controllers/mixins/ems_common/core.rb

```diff
@@ -0,0 +1,76 @@
+module Mixins
+  module EmsCommon
+    module Core
+      def process_emss(emss, task)
+        emss, _emss_out_region = filter_ids_in_region(emss, "Provider")
+        assert_rbac(model, emss)
+
+        return if emss.empty?
+
+        if task == "destroy"
+          model.where(:id => emss).order("lower(name)").each do |ems|
+            id = ems.id
+            ems_name = ems.name
+            audit = {:event        => "ems_record_delete_initiated",
+                     :message      => "[#{ems_name}] Record delete initiated",
+                     :target_id    => id,
+                     :target_class => model.to_s,
+                     :userid       => session[:userid]}
+            AuditEvent.success(audit)
+          end
+          model.destroy_queue(emss)
+          add_flash(n_("Delete initiated for %{count} %{model} from the %{product} Database",
+                       "Delete initiated for %{count} %{models} from the %{product} Database", emss.length) %
+            {:count   => emss.length,
+             :product => Vmdb::Appliance.PRODUCT_NAME,
+             :model   => ui_lookup(:table => table_name),
+             :models  => ui_lookup(:tables => table_name)}) if @flash_array.nil?
+        elsif task == "pause_ems" || task == "resume_ems"
```

Comment:
28 ... 43 should be a new method

Reply:
Yes, it should be :wink: but definitely not in this PR, if you look at the commits, this is just reorganizing some pieces of code, not inventing anything new.